src/dataAPI/jupyter/PythonExecutor.ts

```diff
@@ -267,6 +269,87 @@ export class PythonPandasExecutor {
         }
     }
 
+    public async textFormat(
+        dfName: string,
+        colName: string,
+        type:string,
+    ): Promise<TextFact> {
+        try {
+            if (type == "N") {
+                const describe_code = `print(${dfName}["${replaceSpecial(
+                    colName
+                )}"].describe().to_json())`;
+                const res = await this.executeCode(describe_code);
+                const content = res['content']; // might be null
+                const json_res = JSON.parse(content[0]?.replace(/'/g, '')); // remove single quotes bc not JSON parseable
+                const min = parseFloat(json_res['min'])
+                const q25 = parseFloat(json_res['25%'])
+                const q50 = parseFloat(json_res['50%'])
+                const q75 = parseFloat(json_res['75%'])
+                const max = parseFloat(json_res['max'])
+                const mean= parseFloat(json_res['mean'])
+                return {
+                    name: colName,
+                    description: colName+ " has a minimum of "+(min.toString()) + " and maximum of " + 
+                    (max.toString()) + " with a median of " + q50.toString() +". It has a mean of " + mean.toString() + ".",
+                    extra: undefined
+                };
+            } else if (type == "C") {
+                const info = this.getCatMeta(dfName,colName);
```

Comment:
Move the await to here so that you dont have to await info each time like 

`const info = await this.getCatMeta(dfName, colName)`